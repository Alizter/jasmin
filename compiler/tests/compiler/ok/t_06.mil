// FIXME: parser ignores comments
//        => we inline to test strip_comments

fn bar() {
}

fn foo(x, y : stack u64, z : reg bool) -> stack u64 {
  w : stack u64;
  w = x;
  w += x;
  if (w = 5) {
    z, x += w;
    z, x += y;
  }
  bar();
  return x;
}

/*
START:CMD
ARG="inline[foo],strip_comments[foo],print[input][types,fun=foo]"
END:CMD
*/
