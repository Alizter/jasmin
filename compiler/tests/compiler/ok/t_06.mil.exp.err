Processed file tests/compiler/ok/t_06.mil
