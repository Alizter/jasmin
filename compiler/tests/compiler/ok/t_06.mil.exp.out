inlining all calls in foo function(s)
   REMOVED_TIME
stripping comments in foo function(s)
   REMOVED_TIME
>> input:
fn foo(x : stack u64, y : stack u64, z : reg bool) -> stack u64 {
  w : stack u64
  (w : stack u64) = (x : stack u64);
  (w : stack u64) += (x : stack u64);
  if (w : stack u64 = 5) {
    (z : reg bool), (x : stack u64) += (w : stack u64);
    (z : reg bool), (x : stack u64) += (y : stack u64);
  } else {
    
  }
  
  return (x : stack u64);
}


total transformation time: REMOVED_TIME

total time (with save/print): REMOVED_TIME
